include/deal.II/particles/particle.h

```diff
@@ -235,6 +235,25 @@ namespace Particles
     void
     write_data(void *&data) const;
 
+
+    /**
+     * Update all of the data associated with a particle : id,
+     * location, reference location and, if any, properties by using a
+     * data array. The array is expected to be large enough to take the data,
+     * and the void pointer should point to the first entry of the array to
+     * which the data should be written. This function is meant for
+     * de-serializing the particle data without requiring that a new Particle
+     * class be built. This is used in the ParticleHandler, to update the
+     * ghost particles without de-allocating and re-allocating memory.
+     *
+     * @param[in] data A pointer to a memory location from which
+     * to read the information that completely describes a particle. This
+     * class then de-serializes its data from this memory location and
+     * advance the pointer accordingly.
+     */
+    void
+    update_particle_data(const void *&data);
```

Comment:
At a closer inspection, I realize that you do update the pointer, which is why you pass it by reference. Could you please mark the param as `@param[in/out]` to highlight this? (It is so easy to get confused by `const`ness of pointers)?